habit/models.py

```diff
@@ -0,0 +1,34 @@
+from django.db import models
+from datetime import datetime
+from django.contrib.auth.models import AbstractBaseUser
+# Create your models here.
+
+
+class User(AbstractBaseUser):
```

Comment:
Inheriting from AbstractBaseUser is going to give you some problems with your User model -- you want to inherit instead from AbstractUser.